module/VuFind/src/VuFind/ILS/Driver/KohaRest.php

```diff
@@ -2449,21 +2470,32 @@ protected function mapRenewalBlockReason($reason, $itype)
     }
 
     /**
-     * Return a location for a Koha item
+     * Return a location (branch or shelving) for a Koha item
      *
      * @param array $item Item
      *
      * @return string
      */
     protected function getItemLocationName($item)
     {
-        $libraryId = (!$this->useHomeLibrary && null !== $item['holding_library_id'])
-            ? $item['holding_library_id'] : $item['home_library_id'];
-        $name = $this->translateLocation($libraryId);
-        if ($name === $libraryId) {
-            $libraries = $this->getLibraries();
-            $name = isset($libraries[$libraryId])
+        $name = '';
+        if ($this->config['ItemTypeRenewalBlockMappings']['Location'] == 'Branch') {
+            $libraryId = (!$this->useHomeLibrary && null !== $item['holding_library_id'])
+                ? $item['holding_library_id'] : $item['home_library_id'];
+            $name = $this->translateLocation($libraryId);
+            if ($name === $libraryId) {
+                $libraries = $this->getLibraries();
+                $name = isset($libraries[$libraryId])
                 ? $libraries[$libraryId]['name'] : $libraryId;
+            }
+        } elseif ($this->config['ItemTypeRenewalBlockMappings']['Location'] == 'Shelving') {
```

Comment:
Maybe the location setting should be set to a variable before this if/else so that you don't have to repeat the default value checking here. Or maybe the whole structure should be a switch to make it easier to add more cases in future.